pkg/authorization/api/helpers.go

```diff
@@ -38,6 +61,26 @@ func ExpandResources(rawResources sets.String) sets.String {
 	return ret
 }
 
+func needsNormalizing(in string) bool {
+	// check to see if we conform to common cases for case
+	// this looks insane.  Follow me here and blame Jordan
+	// strings can be cast to []byte for free.  We range over the bytes to avoid
+	// the conversion to runes that happens when ranging over strings.  This results in
+	// zero allocations here.
+	for _, currByte := range []byte(in) {
```

Comment:
> edit: sorry, condition was backwards... it was the memory allocation patterns of IsLower() I was interested in

Alright, since you've dug deep and I haven't.  Does it think '/', '-', or '.' are lowers of anything?